source/d3d11/d3d11_device_context.cpp

```diff
@@ -16,23 +16,254 @@ void D3D11DeviceContext::log_drawcall(UINT vertices)
 		_draw_call_tracker.log_drawcalls(_active_depthstencil.get(), 1, vertices);
 	}
 }
+
+com_ptr<ID3D11Texture2D> D3D11DeviceContext::copy_depth_texture(D3D11_TEXTURE2D_DESC texture_desc, ID3D11Texture2D *depth_texture)
+{
+	com_ptr<ID3D11Texture2D> depth_texture_copy = nullptr;
+
+	switch (texture_desc.Format)
+	{
+		case DXGI_FORMAT_R16_TYPELESS:
+		case DXGI_FORMAT_D16_UNORM:
+			texture_desc.Format = DXGI_FORMAT_R16_TYPELESS;
+			break;
+		case DXGI_FORMAT_R32_TYPELESS:
+		case DXGI_FORMAT_D32_FLOAT:
+			texture_desc.Format = DXGI_FORMAT_R32_TYPELESS;
+			break;
+		default:
+		case DXGI_FORMAT_R24G8_TYPELESS:
+		case DXGI_FORMAT_D24_UNORM_S8_UINT:
+			texture_desc.Format = DXGI_FORMAT_R24G8_TYPELESS;
+			break;
+		case DXGI_FORMAT_R32G8X24_TYPELESS:
+		case DXGI_FORMAT_D32_FLOAT_S8X24_UINT:
+			texture_desc.Format = DXGI_FORMAT_R32G8X24_TYPELESS;
+			break;
+	}
+
+	texture_desc.BindFlags = D3D11_BIND_DEPTH_STENCIL | D3D11_BIND_SHADER_RESOURCE;
+
+	HRESULT hr = _device->CreateTexture2D(&texture_desc, nullptr, &depth_texture_copy);
+
+	if (FAILED(hr))
+	{
+		LOG(ERROR) << "Failed to create depth texture copy! HRESULT is '" << std::hex << hr << std::dec << "'.";
+		return nullptr;
+	}
+
+	this->CopyResource(depth_texture_copy.get(), depth_texture);
+
+	return depth_texture_copy;
+}
+
+ID3D11DepthStencilView *D3D11DeviceContext::copy_depthstencil(ID3D11DepthStencilView *depthStencilView)
+{
+	ID3D11DepthStencilView *depthStencilView_copy = nullptr;
+
+	D3D11_TEXTURE2D_DESC texdesc;
+	com_ptr<ID3D11Resource> resource;
+	com_ptr<ID3D11Texture2D> texture;
+	com_ptr<ID3D11Texture2D> depthstencil_texture;
+	depthStencilView->GetResource(&resource);
+	if (FAILED(resource->QueryInterface(&texture)))
+	{
+		return nullptr;
+	}
+	texture->GetDesc(&texdesc);
+
+	depthstencil_texture = texture.get();
+
+	_draw_call_tracker.set_depth_texture(depthstencil_texture.get());
+
+	D3D11_DEPTH_STENCIL_VIEW_DESC dsvdesc = {};
+	dsvdesc.ViewDimension = D3D11_DSV_DIMENSION_TEXTURE2D;
+
+	switch (texdesc.Format)
+	{
+		case DXGI_FORMAT_R16_TYPELESS:
+			dsvdesc.Format = DXGI_FORMAT_D16_UNORM;
+			break;
+		case DXGI_FORMAT_R32_TYPELESS:
+			dsvdesc.Format = DXGI_FORMAT_D32_FLOAT;
+			break;
+		case DXGI_FORMAT_R24G8_TYPELESS:
+			dsvdesc.Format = DXGI_FORMAT_D24_UNORM_S8_UINT;
+			break;
+		case DXGI_FORMAT_R32G8X24_TYPELESS:
+			dsvdesc.Format = DXGI_FORMAT_D32_FLOAT_S8X24_UINT;
+			break;
+		default:
+			dsvdesc.Format = texdesc.Format;
+			break;
+	}
+
+	HRESULT hr = _device->CreateDepthStencilView(texture.get(), &dsvdesc, &depthStencilView_copy);
+
+	if (FAILED(hr))
+	{
+		LOG(ERROR) << "Failed to create depth stencil view copy! HRESULT is '" << std::hex << hr << std::dec << "'.";
+		return nullptr;
+	}
+
+	return depthStencilView_copy;
+}
+
+bool D3D11DeviceContext::check_depthstencil(com_ptr<ID3D11Texture2D> texture, D3D11_TEXTURE2D_DESC texture_desc)
+{	
+	screen_dimensions();
```

Comment:
See above. Should better be replaced with:
```c++
const auto runtime = _device->_runtimes.front();
const auto screen_width = static_cast<FLOAT>(runtime->frame_width());
const auto screen_height = static_cast<FLOAT>(runtime->frame_height());
```

Reply:
changed